### ISSUE TITLE
fix `try` setting of `$_error`? And turn `$_error` into a List to append to?

### ISSUE THREAD

**Issue by bar-g** [oilshell/oil#1830]
I noticed the first try result is carried on

```
ysh ysh-0.20.0$ try {  echo $[toJ8( /d+/ )] }
ysh ysh-0.20.0$ = _error

(Dict 0x7f4512555688)   {"status":4,"message":"Can't serialize object of type Eggex"}

ysh ysh-0.20.0$ try {  echo Hi! }

Hi!
ysh ysh-0.20.0$ = _error

(Dict 0x7f4512555688)   {"status":4,"message":"Can't serialize object of type Eggex"} 


# Oops, error even though try succeeded.
```

Lets produce another error:
```
ysh ysh-0.20.0$ try false

ysh ysh-0.20.0$ = _error

(Dict 0x7f4512555688)   {"status":4,"message":"Can't serialize object of type Eggex"}


# Oops, unchanged, not catching shell exit code?
```

I think it could be good if $_error would be a List of the last try results (first is newest). (Maybe cleared with `error (reset=all)`?)

That could ease first trying/testing/validating several things and iterating over errors afterwards, to report all problems together.

**Comment by andychu** [comment]
Hmm yeah this could be a bug

Let me see --  it could be that we do need to clear both `_status` and `_error` on every `try`

thanks for testing!

**Comment by bar-g** [comment]
Yes, seems like an important half to get the variable update working.

Think it's required for accessing, I don't seem able to access all errors in a consistent way.

I don't succeed in translating a simple loop over a generic command list in into YSH?

How would you do that?
Instead of the `echo $0` in the example, using `echo $[toJ8( /d+/ )]` in YSH:
```
$ for cmd in "false" "echo $0"; do $cmd; echo $?; done
1
bash
0
```

Hopefully there are still some simplifications possible for defining the list or loop, besides solving the error.
In my attempt the command list definition doesn't look like allowing smooth processing of some command listing:
```
ysh ysh-0.20.0$ var list = [ ^(false), ^(echo $[toJ8( /d+/ )]) ]
ysh ysh-0.20.0$ for cmd in (list) { try (cmd); try echo $[_error.status] }
  for cmd in (list) { try (cmd); try echo $[_error.status] }
                                                  ^
[ interactive ]:2: fatal: Dict entry 'status' not found
ysh ysh-0.20.0$ 
```



Why does `try $shell_command` not set `$[_error.status]` in the same way it sets `$_status`? (Together with emptying $[_error.message].)

Not doing that seems to require having to keep track of the command type executed by try, to properly access the results.

It doesn't even seem to be possible to do another round of try to successfully access the available results.

**Comment by bar-g** [comment]
With the idea I asked about in #799 I pictured it could become possible to splice a string list into command literals.

```
var list = :| false 'echo $[toJ8( /d+/ )]' |
for cmd in @^(list) {
   try (cmd)
   echo $[_error.status] 
}
```

**Comment by andychu** [comment]
Yeah the problem here is that we should set `_error` on every `try`, which is discussed on

Just like we set `_status` on every `try`


- #1833

**Comment by andychu** [comment]
Fixed with https://github.com/oilshell/oil/commit/027ef039c329f3acbba56ab4c72c86ec07471a78

**Comment by andychu** [comment]
Released - https://www.oilshell.org/blog/2024/11/release-0.23.0.html
